src/plugins/intel_cpu/src/nodes/executors/shl/shl_eltwise.cpp

```diff
@@ -0,0 +1,234 @@
+                          "] and dst precision: ", dstDescs[0]->getPrecision().to_string(), " is not supported");
+}
+
+bool ShlEltwiseExecutor::isEltwiseAlgorithmSupported(Algorithm algorithm) {
+    if (one_of(algorithm, Algorithm::EltwiseAdd,
+                          Algorithm::EltwiseSubtract,
+                          Algorithm::EltwiseMultiply,
+                          Algorithm::EltwiseDivide,
+                          Algorithm::EltwiseMaximum,
+                          Algorithm::EltwiseMinimum,
+                          Algorithm::EltwiseExp,
+                          Algorithm::EltwiseClamp,
+                          Algorithm::EltwiseRelu,
+                          Algorithm::EltwisePrelu)) {
+        return true;
+    }
+    return false;
+}
+
+bool ShlEltwiseExecutorBuilder::isSupported(const EltwiseAttrs& eltwiseAttrs,
+                                            const std::vector<MemoryDescPtr>& srcDescs,
+                                            const std::vector<MemoryDescPtr>& dstDescs) const {
+    if (!ShlEltwiseExecutor::isEltwiseAlgorithmSupported(eltwiseAttrs.algorithm)) {
+        DEBUG_LOG("Eltwise algorithm ", algToString(eltwiseAttrs.algorithm), " is not supported");
+        return false;
+    }
+
+    constexpr auto supported_prec = ov::element::f32;
+    auto is_precision_supported = [supported_prec](const MemoryDescPtr& desc) { return desc->getPrecision() == supported_prec; };
+    if (!(std::all_of(srcDescs.cbegin(), srcDescs.cend(), is_precision_supported) &&
+          std::all_of(dstDescs.cbegin(), dstDescs.cend(), is_precision_supported))) {
+        DEBUG_LOG("ShlEltwise supports only f32");
+        return false;
+    }
+
+    for (const auto& srcDesc : srcDescs) {
+        csinn_layout_enum supportedLayout = getShlDataLayoutByMemoryDesc(srcDesc);
+        switch (eltwiseAttrs.algorithm) {
+            case Algorithm::EltwisePrelu:
+                // SHL PRelu op only supports these two kinds of layout
+                if (!(supportedLayout == csinn_layout_enum::CSINN_LAYOUT_NC1HWC0 || supportedLayout == csinn_layout_enum::CSINN_LAYOUT_NCHW)) {
+                    DEBUG_LOG("src descriptor layout is unsupported by SHL Prelu op: ", srcDesc->serializeFormat());
+                    return false;
+                }
+                break;
+            default:
+                if (supportedLayout == csinn_layout_enum::CSINN_LAYOUT_NULL) {
+                    DEBUG_LOG("src descriptor layout is unsupported by SHL: ", srcDesc->serializeFormat());
+                    return false;
+                }
+                continue;
+        }
+    }
+    for (const auto& dstDesc : dstDescs) {
+        if (getShlDataLayoutByMemoryDesc(dstDesc) == csinn_layout_enum::CSINN_LAYOUT_NULL) {
+            DEBUG_LOG("dst descriptor layout is unsupported by SHL: ", dstDesc->serializeFormat());
+            return false;
+        }
+    }
```

Comment:
I see that Eltwise node creates only planar layoit desc (nchw) while this code checks doesn't prohibit nhwc layout. So what is really supported by SHL?

Reply:
I think SHL Eltwise node supports only planar layout by creating descriptors with only `nchw` layout. This part of code does not block any other layouts as the eltwise ops in SHL does not constrain the layouts of their input and output (expect `PRelu`). So in the future more layouts can be supported by creating descriptors with different layouts under different situations.